controllers/controller/devworkspacerouting/solvers/basic_solver.go

```diff
@@ -22,20 +22,33 @@ import (
 	"github.com/devfile/devworkspace-operator/pkg/infrastructure"
 )
 
-var routeAnnotations = func(endpointName string) map[string]string {
-	return map[string]string{
-		"haproxy.router.openshift.io/rewrite-target": "/",
-		constants.DevWorkspaceEndpointNameAnnotation: endpointName,
+func appendMap[K, V comparable](dst map[K]V, m map[K]V) {
+	for k, v := range m {
+		dst[k] = v
 	}
 }
 
-var nginxIngressAnnotations = func(endpointName string) map[string]string {
-	return map[string]string{
-		"kubernetes.io/ingress.class":                "nginx",
-		"nginx.ingress.kubernetes.io/rewrite-target": "/",
-		"nginx.ingress.kubernetes.io/ssl-redirect":   "false",
+var routeAnnotations = map[string]string{
+	"haproxy.router.openshift.io/rewrite-target": "/",
+}
+
+var nginxIngressAnnotations = map[string]string{
+	"kubernetes.io/ingress.class":                "nginx",
+	"nginx.ingress.kubernetes.io/rewrite-target": "/",
+	"nginx.ingress.kubernetes.io/ssl-redirect":   "false",
+}
+
+func createAnnotations(endpointName string, routingAnnotations map[string]string, defaultAnnotations map[string]string) map[string]string {
+	annotations := map[string]string{
 		constants.DevWorkspaceEndpointNameAnnotation: endpointName,
 	}
+
+	if routingAnnotations == nil || len(routingAnnotations) == 0 {
+		appendMap(annotations, defaultAnnotations)
+	} else {
+		appendMap(annotations, routingAnnotations)
+	}
+	return annotations
```

Comment:
A better approach for setting a default value if the configuration is unspecified would be similar to what we do in the [config](https://github.com/devfile/devworkspace-operator/blob/e97ad530f84b6529fd85109a83d16aa0a600c158/pkg/config/defaults.go#L53-L78) package -- this avoids needing functions that take an optional value and a default.